sdk/clientcore/core/src/main/java/io/clientcore/core/http/pipeline/OAuthBearerTokenAuthenticationPolicy.java

```diff
@@ -127,6 +166,24 @@ public boolean authorizeRequestOnChallenge(HttpRequest httpRequest, Response<Bin
         return false;
     }
 
+    /**
+     * Handles the authentication challenge in the event a 401 response with a WWW-Authenticate authentication challenge
+     * header is received after the initial request and returns appropriate {@link OAuthTokenRequestContext} to be
+     * used for re-authentication.
+     *
+     * <p>
+     * The default implementation doesn't handle challenges. You can override and your implementation as needed.
```

Comment:
I'm not convinced we want to default to a bad implementation? If we can do anything useful, it's probably better to do nothing at all?

```suggestion
     * The default implementation doesn't handle challenges. You can override in your implementation as needed.
```

Reply:
I'll leave this question to @g2vinay as I was simply making the async code path based on the sync logic.